plugins/modules/x509_certificate_revocation_info.py

```diff
@@ -0,0 +1,365 @@
+)
+
+from ansible_collections.community.crypto.plugins.module_utils.crypto.identify import (
+    identify_pem_format,
+)
+
+MINIMAL_CRYPTOGRAPHY_VERSION = '1.2'
+
+CRYPTOGRAPHY_IMP_ERR = None
+try:
+    import cryptography
+    from cryptography import x509
+    from cryptography.hazmat.backends import default_backend
+    CRYPTOGRAPHY_VERSION = LooseVersion(cryptography.__version__)
+except ImportError:
+    CRYPTOGRAPHY_IMP_ERR = traceback.format_exc()
+    CRYPTOGRAPHY_FOUND = False
+else:
+    CRYPTOGRAPHY_FOUND = True
+
+
+TIMESTAMP_FORMAT = "%Y%m%d%H%M%SZ"
+
+
+class CertificateRevocationInfo(OpenSSLObject):
+    def __init__(self, module):
+        super(CertificateRevocationInfo, self).__init__(
+            module.params['path'] or '',
+            'present',
+            False,
+            module.check_mode,
+        )
+        self.backend = 'cryptography'
+        self.module = module
+
+        self.content = module.params['content']
+        if self.content is not None:
+            self.content = self.content.encode('utf-8')
+
+        self.cert_serial_number = module.params['serial_number']
+        self.cert = None
+
+    def load(self):
+        if self.content is not None or self.module.params['path'] is not None:
+            self.cert = load_certificate(self.path, content=self.content, backend=self.backend)
+            self.cert_serial_number = cryptography_serial_number_of_cert(self.cert)
+        if self.cert_serial_number is None:
+            raise AssertionError('Internal error - no certificate serial number found')
+
+    def generate(self):
+        # Empty method because OpenSSLObject wants this
+        pass
+
+    def dump(self):
+        # Empty method because OpenSSLObject wants this
+        pass
+
+    def _get_ocsp_uri(self):
+        try:
+            ext = self.cert.extensions.get_extension_for_class(x509.AuthorityInformationAccess)
+            for desc in ext.value:
+                if desc.access_method == x509.oid.AuthorityInformationAccessOID.OCSP:
+                    if isinstance(desc.access_location, x509.UniformResourceIdentifier):
+                        return desc.access_location.value
+        except x509.ExtensionNotFound as dummy:
+            pass
+        return None
+
+    def _check_crl(self, result, crl_blob, crl_source):
+        # Decode CRL
+        try:
+            if identify_pem_format(crl_blob):
+                crl = x509.load_pem_x509_crl(crl_blob, default_backend())
+            else:
+                crl = x509.load_der_x509_crl(crl_blob, default_backend())
+        except Exception as e:
+            self.module.fail_json(msg='Error while decoding CRL from {1}: {0}'.format(e, crl_source))
+
+        # Check revoced certificates
+        if 'crl_contained' not in result:
+            result['crl_contained'] = False
+            result['crl_record'] = None
+        for cert in crl:
+            if cert.serial_number == self.cert_serial_number:
```

Comment:
In addition for serial number match, to be considered revoked a certificate also must have either:
- The same issuer as the CRL
- The "certificateIssuer" field on the CRL entry set to the same issuer as the certificate (which is one or more entries in the issuer alternate names).

Also,there's sort of a "fall through" for CRL issuer values, where if it's not set, it defaults to whatever the value was for the previous entry in the CRL. So you could have CRL Entries (don't quote me on this):
1. <CertIssuer=NotSet> Defaults to Z, which was the CRL Issuer.
2. <CertIssuer=A> Users Cert Issuer A
3. <CertIssuer=NotSet> Uses Cert Issuer A
4. <CertIssuer=NotSet> Uses Cert Issuer A
5. <CertIssuer=B> Uses Cert Issuer B
6. <CertIssuer=NotSet> Uses Cert Issuer B

Relevant section from RFC5280 https://tools.ietf.org/html/rfc5280
> 5.3.3.  Certificate Issuer
> This CRL entry extension identifies the certificate issuer associated
>    with an entry in an indirect CRL, that is, a CRL that has the
>    indirectCRL indicator set in its issuing distribution point
>    extension.  When present, the certificate issuer CRL entry extension
>    includes one or more names from the issuer field and/or issuer
>    alternative name extension of the certificate that corresponds to the
>    CRL entry.  If this extension is not present on the first entry in an
>    indirect CRL, the certificate issuer defaults to the CRL issuer.  On
>    subsequent entries in an indirect CRL, if this extension is not
>    present, the certificate issuer for the entry is the same as that for
>    the preceding entry.  This field is defined as follows:
> 
>    id-ce-certificateIssuer   OBJECT IDENTIFIER ::= { id-ce 29 }
> 
>    CertificateIssuer ::=     GeneralNames
> 
>    Conforming CRL issuers MUST include in this extension the
>    distinguished name (DN) from the issuer field of the certificate that
>    corresponds to this CRL entry.  The encoding of the DN MUST be
>    identical to the encoding used in the certificate.
> 
>    CRL issuers MUST mark this extension as critical since an
>    implementation that ignored this extension could not correctly
>    attribute CRL entries to certificates.  This specification RECOMMENDS
>    that implementations recognize this extension.